keps/0020-manual-plan-execution.md

```diff
@@ -0,0 +1,180 @@
+
+#### Pros
+
+Very easy to implement considering the current state of the code
+
+#### Cons
+
+No way to start a plan with just pure kubectl - this is a reply from sig-cli channel where I posed that question (`It doesn't make sense to update the status manually, it should reflect the current state in the cluster. So no, you cannot set it though kubectl`)
+
+### Expose custom HTTP API from controller-manager to allow receiving commands
+
+Manager binary with controllers (that is run within the pod in the cluster) would in this case have to also expose HTTP API and we’ll have to expose a kubernetes service with this custom API. All clients (CLI/user scripts) will talk to this API every time they want to start a new plan. The API could be as simple as 
+
+HTTP POST `/start?planName=deploy`
+
+The validation whether a plan can be started or not would have to be part of the HTTP endpoint implementation.
+
+#### Pros
+
+The complexity of the implementation is still pretty managable and lower than having a full-featured sub-resource
+
+#### Cons
+
+Have to maintain a separate API (with API documentation)
+It is a non-kubernetes API, so no clients for various languages and no kubectl support (although if we keep the API that simple, almost a CURL can do the trick)
+More complex manager pod deployment with additional service
+How do we manage RBAC?
+
+### Have a separate CRD just for maintaining commands to execute a plan
+
+This would basically mean going half-way back to state pre KEP-18 where we had a separate CR that was created for every plan that was executed. In this implementation we could have a much leaner CRD capturing only the requests to execute a plan with very lean status that does not duplicate the instance's status. Could be as simple as:
+
+```
+apiVersion: kudo.dev/v1beta1
+kind: PlanExecutionRequest
+Spec:
+  planName: deploy
+  Instance: # this will be OwnerReference
+    Name: my-instance
+    Namespace: some-namespace
+  Parameters:
+    some-instance-param: my override
+Status:
+  Status: Accepted # one of Accepted, Rejected, Cancelled (added in the future)
+```
+
+The question is whether we would add a Status into that CRD. If not, it would be hard to tell the client if the execution was even possible and how it ended (in comparison to the API solution where we can validate as part of HTTP response and even return some unique ID for tracking). If we decide to include status, we’re back in pre-KEP-18 world where keeping those two Statuses in sync is hard in kubernetes world with no transactions and both CRDs being reconciled in their own loops.
```

Comment:
Is this paragraph still up-to-date? I thought we agreed to strictly separate between `Instance.Status` which is about Instance plans and `PlanExecutionRequest.Status` which is about *request* itself.

Reply:
yep, will update, thanks